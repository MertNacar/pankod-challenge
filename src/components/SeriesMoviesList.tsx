import React from "react";
import SeriesMoviesCard from "./SeriesMoviesCard";
interface IProp {
  values: Array<Object>;
  classList: string;
}

const SeriesMoviesList: React.FC<IProp> = props => {
  let list = props.values.map((item: any, index: number) => {
    return (
      <span key={index} className={props.classList}>
        <SeriesMoviesCard
          title={item.title}
          image={item.images["Poster Art"].url}
          alt={item.title}
        />
      </span>
    );
  });
  return <>{list}</>;
};

export default SeriesMoviesList;
