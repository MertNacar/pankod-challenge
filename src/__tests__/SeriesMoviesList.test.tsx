import React from "react";
import { shallow } from "enzyme";
import SeriesMoviesList from "../components/SeriesMoviesList";
import mockData from '../../public/feed/test/mock_test.json'

describe("SeriesMoviesList Component", () => {
  const wrapper = shallow(
    <SeriesMoviesList values={mockData} classList="pr-3 pt-3 pb-3" />
  );

  it("is exists", () => {
    expect(wrapper.exists()).toBeTruthy();
  });

  it("return five items", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("return an empty array", () => {
    const wrapperEmpty = shallow(
      <SeriesMoviesList values={[]} classList="pr-3 pt-3 pb-3" />
    );

    expect(wrapperEmpty).toMatchSnapshot("SeriesMoviesList");
  });
});

