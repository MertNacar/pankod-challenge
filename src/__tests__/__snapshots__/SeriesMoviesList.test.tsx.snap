// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesMoviesList Component return an empty array: SeriesMoviesList 1`] = `<Fragment />`;

exports[`SeriesMoviesList Component return five items 1`] = `
<Fragment>
  <span
    className="pr-3 pt-3 pb-3"
    key="0"
  >
    <SeriesMoviesCard
      alt="Wolf Creek"
      image="https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg"
      title="Wolf Creek"
    />
  </span>
  <span
    className="pr-3 pt-3 pb-3"
    key="1"
  >
    <SeriesMoviesCard
      alt="No Activity"
      image="https://streamcoimg-a.akamaihd.net/000/106/36/10636-PosterArt-9add943c5e62c2d89f6d31458d33508a.jpg"
      title="No Activity"
    />
  </span>
  <span
    className="pr-3 pt-3 pb-3"
    key="2"
  >
    <SeriesMoviesCard
      alt="Billions"
      image="https://streamcoimg-a.akamaihd.net/000/117/25/11725-PosterArt-deecf8dbd786dfa2d964413b0bf83726.jpg"
      title="Billions"
    />
  </span>
  <span
    className="pr-3 pt-3 pb-3"
    key="3"
  >
    <SeriesMoviesCard
      alt="Better Call Saul"
      image="https://streamcoimg-a.akamaihd.net/000/175/7/1757-PosterArt-fc0e2a5b18d4a662717f3cbb1827871d.jpg"
      title="Better Call Saul"
    />
  </span>
  <span
    className="pr-3 pt-3 pb-3"
    key="4"
  >
    <SeriesMoviesCard
      alt="Younger"
      image="https://streamcoimg-a.akamaihd.net/000/143/24/14324-PosterArt-a0168953a3b93681dbd60c7bcc50a0af.jpg"
      title="Younger"
    />
  </span>
</Fragment>
`;
